lenskit-svd/src/main/java/org/grouplens/lenskit/mf/funksvd/FunkSVDItemScorer.java

```diff
@@ -165,34 +164,34 @@ public void score(long user, @Nonnull MutableSparseVector scores) {
         scores.set(estimates);
 
         if (!ratings.isEmpty() && rule != null) {
-            AVector updated = Vector.create(uprefs);
+            //RealVector updated = uprefs.copy();
             for (int f = 0; f < featureCount; f++) {
-                trainUserFeature(user, updated, ratings, estimates, f);
+                trainUserFeature(user, uprefs, ratings, estimates, f); // changed updated to uprefs
```

Comment:
We still want to use `updated` - we don't want to update the vector stored in the model.